Framework/DataHandling/inc/MantidDataHandling/CompressEventAccumulator.h

```diff
@@ -0,0 +1,72 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidDataHandling/DllConfig.h"
+#include "MantidDataObjects/EventList.h"
+
+#include <vector>
+
+namespace Mantid {
+namespace DataHandling {
+
+enum class CompressBinningMode { LINEAR, LOGARITHMIC };
+
+/** CompressEventAccumulator : TODO: DESCRIPTION
+ */
+class MANTID_DATAHANDLING_DLL CompressEventAccumulator {
+public:
+  // TODO parameter for expected number of events
+  CompressEventAccumulator(std::shared_ptr<std::vector<double>> histogram_bin_edges, const double divisor,
+                           CompressBinningMode bin_mode);
```

Comment:
1. We carry around the `bin_mode` and `divisor` separately so the bin to put the event in can be calculated rather than searched for. This is significantly faster than sorting then finding or just using a binary search. `EventList` has methods for finding the bin via math. [EventList::findLinearBin](https://github.com/mantidproject/mantid/blob/main/Framework/DataObjects/src/EventList.cpp#L2319) and [EventList::findLogBin](https://github.com/mantidproject/mantid/blob/main/Framework/DataObjects/src/EventList.cpp#L2348) are the methods.
2. Log binning isn't base-10. It is actually that `delta_tof/tof = konstant`. Then `tof[i+1] = tof[i} * konstant`.